### ISSUE TITLE
Task ID filtering broken

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#5186]
Discovered while looking into #5185 

Example:
```ini
[task parameters]
    m = 1..5
[scheduling]
    cycling mode = integer
    initial cycle point = 1
    [[graph]]
        P1 = """
           f<m-1> => f<m>
        """
        R1 = a
        R1/3 = "a[1] => f<m=5>"
[runtime]
    [[a]]
    [[f<m>]]
    [[f<m=1>]]
        script = false
    [[f<m=5>]]
        pre-script = sleep 10
```
![graph](https://user-images.githubusercontent.com/2362137/194805587-e532ce50-800d-4a6d-9320-c277bbb511c9.png)

Run till stalled:

![shot](https://user-images.githubusercontent.com/2362137/194805746-7a40b970-36a7-40ea-a681-15f89793dfeb.png)

Note the scheduler now has `3/f_m5` (partially satisfied) in the hidden task pool (but not visible in the datastore ... hence the hidden bit).

Now try to trigger all failed tasks:
```console
$ cylc trigger 'bug//*/*:failed'
```

This results in `*/f_m1:failed` AND `3/f_m5` triggering, which implies the ID filter is returning the entire current (main and hidden) task pool, not just the incomplete failed tasks.

**Comment by oliver-sanders** [comment]
It looks like the `:failed` qualifier is being stripped by `cylc trigger`, long before the ID makes it to the ID filter code.

**Comment by hjoliver** [comment]
Closed by #5188
